### PR TITLE
Add max depth parameter for recursive macro

### PR DESCRIPTION
**Pull request by Kyzegs**

### CODE REVIEW

File: src/Macros/Recursive.php [hunk 1]
```diff
@@ -15,10 +18,14 @@ class Recursive
 {
     public function __invoke()
     {
-        return function (): Collection {
-            return $this->map(function ($value) {
+        return function (float $maxDepth = INF, int $depth = 0): Collection {
```
Comment (freekmurze) [inline src/Macros/Recursive.php:21]: Since in the old code parameters weren't required, and now there are, this is a breaking change.

**Comment by freekmurze** [comment]
I'm going to pass on this as this PR contains a breaking change. Feel free to adapt this PR to make it non breaking. Also document any new behaviour in the readme.

File: src/Macros/Recursive.php (hunk 1 above)
Comment (Kyzegs) [reply src/Macros/Recursive.php:21]: The parameters have a default value. Any existing calls to the recursive method without any parameters continue to work as intended.

Comment (freekmurze) [reply src/Macros/Recursive.php:21]: Right! I must have needed some coffee the moment I review the code previously.

**Comment by freekmurze** [comment]
Could you add a good example on how to use the new optional parameters in the readme?

**Comment by Kyzegs** [comment]
I can do that! I'll also make sure to fix the tests.

**Comment by Kyzegs** [comment]
Should be good to go now @freekmurze!

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**